scripts/gen_flatbuffer.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+echo "Running gen_flatbuffers.sh OSTYPE is $OSTYPE"
+ROOT=$(pwd)
+FF_LOCATION="$ROOT/third_party/flatbuffers"
+cd "$FF_LOCATION" || exit
+mkdir build
+cd build || exit
+py() { command python "$@"; }
+if [[ "$OSTYPE" == "darwin"* ]]; then
+   cmake -G "Xcode" -DCMAKE_BUILD_TYPE=Release -DCMAKE_OSX_ARCHITECTURES=x86_64 ..
+   cmake --build . --target flatc
+   # for some reason on mac the binary appears in a different directory...
+   FLATC=./Debug/flatc
+else
```

Comment:
Can you please explain why Darwin build needs to be special-cased? This would not work on Darwin systems where only Xcode tools are installed rather than xcode. And what is wrong with regular `make`/`ninja` build? It would be nice to pass generator user selected when running top level CMAKE? (Should be deinfed as top level `CMAKE_GENERATOR` variable)

Reply:
I copied the build command from here: https://google.github.io/flatbuffers/flatbuffers_guide_building.html

Q: What is the different between`-G "Xcode"` and `-G "xcode"`?

Q2: Along with previous comment should I change it to `-G $CMAKE_GENERATOR`?

As for why it appears as ./Debug/flatc, I don't know... I got CI failure on file not found then I printed out generated files and saw the path in ./Debug/flatc. If you know how to control this please let me know.